[BUG]: Qwen3-235B-A22B-FP8 TRT-LLM disagg recipe crashes on H200 (DEEPGEMM dtype assertion / NVRTC compile fail)

### Describe the Bug

The recipe at [`recipes/qwen3-235b-a22b-fp8/trtllm/disagg/deploy.yaml`](https://github.com/ai-dynamo/dynamo/blob/main/recipes/qwen3-235b-a22b-fp8/trtllm/disagg/deploy.yaml) on `main` does not run on H200 as written. Workers enter `CrashLoopBackOff` at engine warmup under **two different runtime images**, with two distinct failure modes. Stripping the `moe_config` block (i.e. the v1.0.1 recipe shape with main's image tag) produces a working deployment.

### Environment

- **Hardware**: 8× H200 141GB on Nebius (Teleport cluster `dynamo-nebius-2`). Driver mix: 590.48.01 on 6 nodes, 595.58.03 on 2 nodes (CUDA 13.2).
- **Dynamo**: `main` @ `6378793939` (recipe from commit `956f93f675`)
- **Operator**: `nvcr.io/nvidia/ai-dynamo/kubernetes-operator:1.0.1` (overriding chart default 1.1.0 which isn't published to NGC)
- **Config**: recipe-as-written — 6 prefill × TP2 + 1 decode × TP4 = 16 GPU. `moe_config: { backend: DEEPGEMM, max_num_tokens: 8192 }` on both prefill + decode.

### Steps to Reproduce

1. `git clone https://github.com/ai-dynamo/dynamo && cd dynamo`
2. Provision namespace + HF / NGC secrets on an H200 K8s cluster
3. Create `model-cache` PVC (RWX, 300Gi) and download `Qwen/Qwen3-235B-A22B-FP8` weights
4. `kubectl apply -n <ns> -f recipes/qwen3-235b-a22b-fp8/trtllm/disagg/deploy.yaml`
5. Workers never reach Ready; RESTARTS increment every ~90s, CrashLoopBackOff
6. `kubectl logs <prefill-pod> --previous | grep -E "Assertion|NVRTC"` shows the failure

**Workaround**: delete the 3-line `moe_config` block from both `prefill.yaml` and `decode.yaml` ConfigMaps, re-apply, delete crashlooping pods. Next generation comes up clean and `/v1/models` responds.

### Expected Behavior

Recipe deploys cleanly on H100/H200 (per [recipe README](https://github.com/ai-dynamo/dynamo/blob/main/recipes/qwen3-235b-a22b-fp8/README.md) which says "16× H100/H200").

### Actual Behavior

**Failure Mode 1** — `tensorrtllm-runtime:1.0.0` (recipe's pinned tag) + DEEPGEMM backend:

\`\`\`
File .../tensorrt_llm/_torch/modules/fused_moe/fused_moe_deepgemm.py:694 in run_moe
    deepgemm_fp8_group_blockwise_gemm(...)
File .../fused_moe_deepgemm.py:349 in deepgemm_fp8_group_blockwise_gemm
    deep_gemm.fp8_m_grouped_gemm_nt_masked((a, sfa), (b, sfb), ...)
RuntimeError: Assertion error
  (_deps/deepgemm-src/csrc/apis/../jit_kernels/impls/../heuristics/../../utils/layout.hpp:49):
  sfa_dtype == torch::kFloat and sfb_dtype == torch::kFloat
\`\`\`

Scale-factor tensors arrive in the wrong dtype to DeepGEMM's blockwise fp8 kernel. Crash during `PyExecutor._run_autotuner_warmup`.

**Failure Mode 2** — `tensorrtllm-runtime:1.1.0-dev.2` (latest pre-release in NGC, TRT-LLM 1.3.0rc9, nvcc 13.1). Tried three variants (`backend: DEEPGEMM`, `backend: TRTLLM`, `moe_config` removed entirely); all three fail identically:

\`\`\`
[TensorRT-LLM][ERROR] NVRTC compilation failed    (×700+ lines, one per kernel variant)
RuntimeError: NVRTC compilation failed
  in tensorrt_llm/_torch/pyexecutor/model_engine.py:627 (warmup)
\`\`\`

Loads weights successfully (2452 shards, ~60s), allocates KV cache, sets up NIXL/UCX, then fails at JIT kernel compilation during autotuner warmup. No specific compile error surfaces — only the generic "failed" message is emitted per kernel variant.

**Working configuration**: `tensorrtllm-runtime:1.0.0` + `moe_config` block **completely removed** from both prefill/decode configs (the v1.0.1 recipe shape with main's image tag). Workers come Ready in ~5 min, `/v1/chat/completions` generates correctly. Baseline benchmark (`perf.yaml` unchanged, ISL=4000, OSL=200, conc=32) completes in 43s at 1484 tok/s system = **92.76 output tokens/sec/GPU** across 16 H200 GPUs.

### Context

The `moe_config: DEEPGEMM` block was added to the disagg recipe on main between v1.0.1 and current HEAD — the diff:

\`\`\`diff
    kv_cache_config:
      enable_block_reuse: true
      free_gpu_memory_fraction: 0.7
      dtype: fp8
    cache_transceiver_config:
      backend: DEFAULT
    cuda_graph_config:
      enable_padding: true
      max_batch_size: 2
    disable_overlap_scheduler: true
    print_iter_log: false
+   moe_config:
+     backend: DEEPGEMM
+     max_num_tokens: 8192
\`\`\`

No runtime image bump accompanied the DEEPGEMM addition — main still references `tensorrtllm-runtime:1.0.0`. That combination is the one that fails.

### Related Issues (surveyed; none are exact matches)

- ai-dynamo/dynamo#3111 — SGLang + DeepSeek-V3 DeepGEMM compile fail, closed (different backend stack)
- NVIDIA/TensorRT-LLM#9021 — Qwen3-Coder-480B + Qwen3-235B-A22B-**Instruct-2507**-FP8 MoE expert-count limit, fixed in PR #7761 / 1.2.0rc2 (different model variant)
- NVIDIA/TensorRT-LLM#6805 — L40 SM89 backend fallback explained: FP8 block-scale MoE → `fp8_block_scale_moe_runner` is SM100+ only; CUTLASS falls through to DeepGEMM on Hopper
- NVIDIA/TensorRT-LLM#5872 — Qwen3-235B H200 TP8+EP8 OOM (distinct; confirms the model ran on H200 in 1.0.0rc1 era)

### Suggested Fix

Three things worth doing:

1. **Short-term recipe fix**: remove the `moe_config: { backend: DEEPGEMM, max_num_tokens: 8192 }` block from both the prefill and decode ConfigMaps in the `trtllm/disagg` and `trtllm/agg` recipes. Fall back to the default MoE backend (which worked pre-addition).
2. **Or: pin a runtime image where DEEPGEMM works**. The recipe change was made without bumping the image tag. If a TRT-LLM 1.2.0rc2+ runtime image exists in NGC, pin that instead of `1.0.0`.
3. **CI smoke**: the README says "H100/H200" but the recipe regressed silently on H200. A minimal deploy + `/v1/models` check on H200 in CI would catch this.

--- a/recipes/qwen3-235b-a22b-fp8/README.md
+++ b/recipes/qwen3-235b-a22b-fp8/README.md
@@ -12,7 +12,7 @@ Production-ready deployments for **Qwen3-235B-A22B** (MoE model with 22B active 
 ## Prerequisites
 
 1. **Dynamo Platform installed** — See [Kubernetes Deployment Guide](../../docs/kubernetes/README.md)
-2. **GPU cluster** with H100/H200 GPUs (high memory recommended)
+2. **GPU cluster** with Blackwell GPUs (B100/B200; SM100+) — see [Hardware Requirements](#hardware-requirements)
 3. **HuggingFace token** with access to Qwen models
 
 ## Quick Start
@@ -62,12 +62,16 @@ curl http://localhost:8000/v1/chat/completions \
 
 ## Hardware Requirements
 
-This is a large MoE model requiring significant GPU resources:
+This recipe uses `moe_config.backend: DEEPGEMM`, which requires **Blackwell GPUs (SM100+, e.g. B100/B200)**.
+DeepGEMM's FP8 grouped-GEMM kernels are designed for SM100/SM103 only and will crash on Hopper (SM90).
+
+> **Note:** To run on Hopper (H100/H200), remove the `moe_config` block from the ConfigMaps in
+> `deploy.yaml`. This falls back to the default MoE backend at a modest throughput reduction.
 
 | Configuration | GPUs | Min GPU VRAM (Total) |
 |--------------|------|----------------------|
-| Aggregated | 16x H100/H200 | ~1.3TB |
-| Disaggregated | 16x H100/H200 | ~1.3TB |
+| Aggregated | 16x B100/B200 | ~1.3TB |
+| Disaggregated | 16x B100/B200 | ~1.3TB |
 
 ## Notes
 

--- a/recipes/qwen3-235b-a22b-fp8/README.md
+++ b/recipes/qwen3-235b-a22b-fp8/README.md
@@ -65,8 +65,8 @@ curl http://localhost:8000/v1/chat/completions \
 This recipe uses `moe_config.backend: DEEPGEMM`, which requires **Blackwell GPUs (SM100+, e.g. B100/B200)**.
 DeepGEMM's FP8 grouped-GEMM kernels are designed for SM100/SM103 only and will crash on Hopper (SM90).
 
-> **Note:** To run on Hopper (H100/H200), remove the `moe_config` block from the ConfigMaps in
-> `deploy.yaml`. This falls back to the default MoE backend at a modest throughput reduction.
+> **Note:** To run on Hopper (H100/H200, SM90), remove the `moe_config` block from the ConfigMaps in
+> `trtllm/agg/deploy.yaml` and `trtllm/disagg/deploy.yaml`. This falls back to the default MoE backend at a modest throughput reduction.
 
 | Configuration | GPUs | Min GPU VRAM (Total) |
 |--------------|------|----------------------|

--- a/recipes/qwen3-235b-a22b-fp8/trtllm/agg/deploy.yaml
+++ b/recipes/qwen3-235b-a22b-fp8/trtllm/agg/deploy.yaml
@@ -24,7 +24,7 @@ data:
       max_batch_size: 128
     disable_overlap_scheduler: false
     print_iter_log: false
-    # DEEPGEMM requires Blackwell (SM100+). Remove this block to run on Hopper (SM90).
+    # DEEPGEMM requires Blackwell (SM100+). Remove moe_config block to run on Hopper (SM90).
     moe_config:
       backend: DEEPGEMM
       max_num_tokens: 8192

--- a/recipes/qwen3-235b-a22b-fp8/trtllm/agg/deploy.yaml
+++ b/recipes/qwen3-235b-a22b-fp8/trtllm/agg/deploy.yaml
@@ -24,6 +24,7 @@ data:
       max_batch_size: 128
     disable_overlap_scheduler: false
     print_iter_log: false
+    # DEEPGEMM requires Blackwell (SM100+). Remove this block to run on Hopper (SM90).
     moe_config:
       backend: DEEPGEMM
       max_num_tokens: 8192

--- a/recipes/qwen3-235b-a22b-fp8/trtllm/disagg/deploy.yaml
+++ b/recipes/qwen3-235b-a22b-fp8/trtllm/disagg/deploy.yaml
@@ -24,7 +24,7 @@ data:
       max_batch_size: 2
     disable_overlap_scheduler: true
     print_iter_log: false
-    # DEEPGEMM requires Blackwell (SM100+). Remove this block to run on Hopper (SM90).
+    # DEEPGEMM requires Blackwell (SM100+). Remove moe_config block to run on Hopper (SM90).
     moe_config:
       backend: DEEPGEMM
       max_num_tokens: 8192
@@ -53,7 +53,7 @@ data:
       max_batch_size: 512
     disable_overlap_scheduler: false
     print_iter_log: false
-    # DEEPGEMM requires Blackwell (SM100+). Remove this block to run on Hopper (SM90).
+    # DEEPGEMM requires Blackwell (SM100+). Remove moe_config block to run on Hopper (SM90).
     moe_config:
       backend: DEEPGEMM
       max_num_tokens: 8192

--- a/recipes/qwen3-235b-a22b-fp8/trtllm/disagg/deploy.yaml
+++ b/recipes/qwen3-235b-a22b-fp8/trtllm/disagg/deploy.yaml
@@ -24,6 +24,7 @@ data:
       max_batch_size: 2
     disable_overlap_scheduler: true
     print_iter_log: false
+    # DEEPGEMM requires Blackwell (SM100+). Remove this block to run on Hopper (SM90).
     moe_config:
       backend: DEEPGEMM
       max_num_tokens: 8192
@@ -52,6 +53,7 @@ data:
       max_batch_size: 512
     disable_overlap_scheduler: false
     print_iter_log: false
+    # DEEPGEMM requires Blackwell (SM100+). Remove this block to run on Hopper (SM90).
     moe_config:
       backend: DEEPGEMM
       max_num_tokens: 8192

